src/request/browser-request.ts

```diff
@@ -40,8 +40,24 @@ export default (context: any, callback: any) => {
   let timedOut = false
 
   // Apply event handlers
-  xhr.onerror = onError
-  xhr.ontimeout = onError
+  xhr.onerror = (event: ProgressEvent) => {
+    onError(
+      new Error(
+        `Request error while attempting to reach ${options.url}${
+          event.lengthComputable ? `(${event.loaded} of ${event.total} bytes transferred)` : ''
+        }`
+      )
```

Comment:
It'd be nice to pass [cause](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause) so the original event can still be accessed:

```suggestion
      new Error(
        `Request error while attempting to reach ${options.url}${
          event.lengthComputable ? `(${event.loaded} of ${event.total} bytes transferred)` : ''
        }`,
        {cause: event}
      )
```